src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -201,161 +203,139 @@ internal async Task<List<PostgresType>> LoadBackendTypes(NpgsqlConnector conn, N
             if (timeout.IsSet)
                 commandTimeout = (int)timeout.CheckAndGetTimeLeft().TotalSeconds;
 
-            using var batch = new NpgsqlBatch(conn)
-            {
-                BatchCommands =
-                {
-                    new("SELECT version()"),
-                    new(GenerateLoadTypesQuery(SupportsRangeTypes, conn.Settings.LoadTableComposites)),
-                    new(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites)),
-                }
-            };
+            var batchQuery = new StringBuilder("SELECT version();")
+                .Append(GenerateLoadTypesQuery(SupportsRangeTypes, conn.Settings.LoadTableComposites))
+                .Append(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites));
 
             if (SupportsEnumTypes)
-                batch.BatchCommands.Add(new(GenerateLoadEnumFieldsQuery(HasEnumSortOrder)));
-
-            batch.Timeout = commandTimeout;
-            batch.AllResultTypesAreUnknown = true;
+                batchQuery.Append(GenerateLoadEnumFieldsQuery(HasEnumSortOrder));
 
+            conn.WriteBuffer.Timeout = TimeSpan.FromSeconds(commandTimeout);
             timeout.CheckAndApply(conn);
-            var reader = async ? await batch.ExecuteReaderAsync() : batch.ExecuteReader();
-            try
-            {
-                var byOID = new Dictionary<uint, PostgresType>();
 
-                // First the PostgreSQL version
-                if (async)
-                {
-                    await reader.ReadAsync();
-                    LongVersion = reader.GetString(0);
-                    await reader.NextResultAsync();
-                }
-                else
+            var q = batchQuery.ToString();
+            Debug.WriteLine($"Type loading query:{Environment.NewLine}{q}");
```

Comment:
This probably slipped in - if we want to log this query, we need a proper NpgsqlLogger (see NpgsqlCommand.LogCommand for how we log regular queries).

It may be best to place a logging in NpgsqlConnector.WriteQuery (called just below), so that we get logging for other simple query commands (e.g. COMMIT).

Reply:
Its a leftover from developing.
I'll remove it.